### PR TITLE
[CIAC-6413][PANOS] Add CRUD commands for security profile groups

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-6413)

## Description
Add CRUD commands for security profile groups in Panorama.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by dantavori — COMMENTED** [review]
looks good so far, will re-review after the demo

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 1]
```diff
@@ -13660,6 +13660,184 @@ def pan_os_delete_tag_command(args: dict) -> CommandResults:
...
+        "type": "config",
+        "action": "delete",
+        "key": API_KEY,
+        "xpath": f"{XPATH_RULEBASE}profile-group/entry[@name='{group_name}']",
+    }
+
+    raw_response = http_request(URL, "GET", params=params)
+    return CommandResults(
+        raw_response=raw_response,
+        readable_output=f'Successfull deleted Security Profile Group: "{group_name}"',
```
Comment (dantavori) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:13837]:
```suggestion
        readable_output=f'Successfully deleted Security Profile Group: "{group_name}"',
```
in all of the new commands

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 2]
```diff
@@ -9364,6 +9364,97 @@ script:
...
+      description: The URL filtering profile.
+      type: String
+    - contextPath: Panorama.ProfileGroup.file-blocking
+      description: The file blocking profile.
+      type: String
+    - contextPath: Panorama.ProfileGroup.data-filtering
+      description: The data filtering profile.
+      type: String
+    - contextPath: Panorama.ProfileGroup.wildfire-analysis
+      description: The wildFire analysis profile.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9400]:
```suggestion
      description: The WildFire analysis profile.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 3]
```diff
@@ -9364,6 +9364,97 @@ script:
...
+      name: anti_spyware_profile
+    - description: The vulnerability protection profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
+      name: vulnerability_protection_profile
+    - description: The URL filtering profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
+      name: URL_filtering_profile
+    - description: The file blocking profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
+      name: file_blocking_profile
+    - description: The data filtering profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
+      name: data_filtering_profile
+    - description: The wildfire analysis profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9423]:
```suggestion
    - description: The WildFire analysis profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command.
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 4]
```diff
@@ -9014,3 +9014,162 @@ There is no context output for this command.
...
+| --- | --- | --- |
+| Panorama.ProfileGroup.name | String | The name of the group. | 
+| Panorama.ProfileGroup.location | String | The security profile group's device group location. | 
+| Panorama.ProfileGroup.virus | String | The antivirus profile. | 
+| Panorama.ProfileGroup.spyware | String | The anti-spyware profile. | 
+| Panorama.ProfileGroup.vulnerability | String | The vulnerability protection profile. | 
+| Panorama.ProfileGroup.url-filtering | String | The URL filtering profile. | 
+| Panorama.ProfileGroup.file-blocking | String | The file blocking profile. | 
+| Panorama.ProfileGroup.data-filtering | String | The data filtering profile. | 
+| Panorama.ProfileGroup.wildfire-analysis | String | The wildFire analysis profile. |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:9046]:
```suggestion
| Panorama.ProfileGroup.wildfire-analysis | String | The WildFire analysis profile. | 
```

File: Packs/PAN-OS/Integrations/Panorama/README.md [hunk 5]
```diff
@@ -9014,3 +9014,162 @@ There is no context output for this command.
...
+| anti_spyware_profile | The anti-spyware profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
+| vulnerability_protection_profile | The vulnerability protection profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
+| URL_filtering_profile | The URL filtering profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
+| file_blocking_profile | The file blocking profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
+| data_filtering_profile | The data filtering profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
+| wildfire_analysis_profile | The wildfire analysis profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/Integrations/Panorama/README.md:9108]:
```suggestion
| wildfire_analysis_profile | The WildFire analysis profile of the group. You can get the possible values for this argument by running the pan-os-get-security-profile command. | Optional | 
```

File: Packs/PAN-OS/ReleaseNotes/2_1_23.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### Palo Alto Networks PAN-OS
+
+- Added the following new commands to manage Panorama security profile group:
+    - ***pan-os-list-security-profile-group***
+    - ***pan-os-create-security-profile-group***
+    - ***pan-os-edit-security-profile-group***
+    - ***pan-os-delete-security-profile-group***
```
Comment (ShirleyDenkberg) [inline Packs/PAN-OS/ReleaseNotes/2_1_23.md:10]:
```suggestion
Added the following new commands to manage a Panorama security profile group:
 - ***pan-os-list-security-profile-group***
 - ***pan-os-create-security-profile-group***
 - ***pan-os-edit-security-profile-group***
 - ***pan-os-delete-security-profile-group***
```

**Comment by ShirleyDenkberg** [comment]
@dantavori @jlevypaloalto Doc review completed.

File: Packs/PAN-OS/Integrations/Panorama/Panorama.yml [hunk 7]
```diff
@@ -9372,6 +9372,97 @@ script:
...
+      predefined:
+      - 'Antivirus Profile'
+      - 'Anti-Spyware Profile'
+      - 'Vulnerability Protection Profile'
+      - 'URL Filtering Profile'
+      - 'File Blocking Profile'
+      - 'Data Filtering  Profile'
+      - 'WildFire Analysis Profile'
+      required: true
+    - description: The value to set in the profile. Set the value "None" if you want to empty the profile value. You can get the possible values for this argument by running the pan-os-get-security-profile command.
```
Comment (yaakovpraisler) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.yml:9451]:
```suggestion
    - description: The value to set in the profile. If you want the profile value to be empty, set the value to "None". You can get the possible values for this argument by running the pan-os-get-security-profile command.
```

File: Packs/PAN-OS/Integrations/Panorama/Panorama.py [hunk 8]
```diff
@@ -13704,6 +13704,243 @@ def pan_os_delete_tag_command(args: dict) -> CommandResults:
...
+
+    params = {
+        "type": "config",
+        "action": "edit",
+        "key": API_KEY,
+    }
+
+    if profile_value.lower() == "none":
+        _, sp_group = pan_os_list_security_profile_groups(args=args)
+        sp_group = sp_group[0]
```
Comment (dantavori) [inline Packs/PAN-OS/Integrations/Panorama/Panorama.py:13897]: check if not empty before

**Review by dantavori — APPROVED** [review]

**Comment by yaakovpraisler** [comment]
Test playbooks are failing to run because [PAN-OS instance issues](https://jira-dc.paloaltonetworks.com/browse/CIAC-9915).
@dantavori We need a force merge.

**Pull request merged**